torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -28,9 +28,20 @@ def get_base(tensor):
 @dataclass
 class ViewInfo:
     base_index: int
-    size: Optional[Sequence[Union[int, torch.SymInt]]] = None
-    stride: Optional[Sequence[Union[int, torch.SymInt]]] = None
-    storage_offset: Optional[int] = None
+    # size, stride, and storage_offset passed to as_strided.
```

Comment:
Can we make ViewInfo a Union[AsStridedInfo, SliceInfo, AliasInfo]? This class is getting a bit big

Reply:
They can all be subclasses of a base ViewInfo class that has a .regenerate_view function on it

Reply:
Maybe add a comment here for "If you need more complicated views in the future, then you're out of luck -- go implement auto_functionalized_v3"

Reply:
i was thinking for one more for subscript since its common

Reply:
Sure, that's fine by me